inventory/inv_scripts/inventory_slot.gd

```diff
@@ -0,0 +1,15 @@
+extends Panel
+
+@onready var item_visual : Sprite2D = $CenterContainer/Panel/item_display
+@onready var amount_text: Label = $CenterContainer/Panel/Label
+
+func update(slot: InvSlot):
+	if !slot.item:
+		item_visual.visible = false
+		amount_text.visible = false
```

Comment:
Consider early return here to avoid a lot of nesting in the else block
```
if ():
  ...
  return
...
```
instead of
```
if ():
  ...
else:
  ...
```